receiver/googlecloudmonitoringreceiver/internal/metrics_conversion.go

```diff
@@ -120,3 +121,96 @@ func (mb *MetricsBuilder) ConvertDeltaToMetrics(ts *monitoringpb.TimeSeries, m p
+	m.SetName(ts.GetMetric().GetType())
+	m.SetUnit(ts.GetUnit())
+	dist := m.SetEmptyHistogram()
+
+	switch metricKind {
+	case metric.MetricDescriptor_CUMULATIVE:
+		dist.SetAggregationTemporality(pmetric.AggregationTemporalityCumulative)
+	case metric.MetricDescriptor_DELTA:
+		dist.SetAggregationTemporality(pmetric.AggregationTemporalityDelta)
+	default:
+		mb.logger.Warn("Unsupported metric kind:", zap.String("Metric", ts.GetMetric().GetType()))
+		return m
+	}
+
+	for _, point := range ts.GetPoints() {
+		dp := dist.DataPoints().AppendEmpty()
+
+		// Set timestamps if valid
+		if point.Interval.StartTime != nil && point.Interval.StartTime.IsValid() {
+			dp.SetStartTimestamp(pcommon.NewTimestampFromTime(point.Interval.StartTime.AsTime()))
+		}
+
+		if point.Interval.EndTime != nil && point.Interval.EndTime.IsValid() {
+			dp.SetTimestamp(pcommon.NewTimestampFromTime(point.Interval.EndTime.AsTime()))
+		} else {
+			mb.logger.Warn("EndTime is invalid for metric:", zap.String("Metric", ts.GetMetric().GetType()))
+		}
+
+		distValue := point.Value.GetDistributionValue()
+		if distValue == nil {
+			mb.logger.Warn("Distribution value is nil for metric:", zap.String("Metric", ts.GetMetric().GetType()))
+			continue
+		}
+
+		dp.SetCount(uint64(distValue.Count))
+
+		// Handle bucket options based on type
+		if bucketOpts := distValue.BucketOptions; bucketOpts != nil {
+			if expBuckets := bucketOpts.GetExponentialBuckets(); expBuckets != nil {
+				// Initialize all buckets with zeros for exponential buckets
+				numBuckets := int(expBuckets.NumFiniteBuckets)
+				dp.BucketCounts().EnsureCapacity(numBuckets)
+				for i := 0; i < numBuckets; i++ {
+					dp.BucketCounts().Append(0)
+				}
+			} else if linearBuckets := bucketOpts.GetLinearBuckets(); linearBuckets != nil {
+				// Initialize all buckets with zeros for linear buckets
+				numBuckets := int(linearBuckets.NumFiniteBuckets)
+				dp.BucketCounts().EnsureCapacity(numBuckets)
+				for i := 0; i < numBuckets; i++ {
+					dp.BucketCounts().Append(0)
+				}
+			} else if explicitBuckets := bucketOpts.GetExplicitBuckets(); explicitBuckets != nil {
+				bounds := explicitBuckets.Bounds
+				dp.ExplicitBounds().EnsureCapacity(len(bounds))
+				for _, bucket := range bounds {
+					dp.ExplicitBounds().Append(bucket)
+				}
+			}
+		}
+
+		// Set bucket counts from the distribution data
+		if len(distValue.BucketCounts) > 0 {
+			// Update existing bucket counts instead of appending
+			for i, count := range distValue.BucketCounts {
+				if i < dp.BucketCounts().Len() {
+					dp.BucketCounts().SetAt(i, uint64(count))
+				} else {
+					// If we somehow have more counts than buckets, append the extras
+					dp.BucketCounts().Append(uint64(count))
+				}
+			}
+		}
+
+		// Set exemplars
+		if exemplars := distValue.Exemplars; exemplars != nil {
+			for _, exemplar := range exemplars {
+				dp.Exemplars().AppendEmpty().SetDoubleValue(float64(exemplar.Value))
```

Comment:
You will want to make sure you get the trace ID and span ID from the SpanContext exemplar attachment. You will also want to set the exemplar's filtered attributes from the DroppedLabels attachment.  See https://github.com/GoogleCloudPlatform/opentelemetry-operations-go/blob/33641345ef4756e48d7011fbb38a5c64ec6b3827/exporter/collector/metrics.go#L967